apps/payments-api/routes/transactions/index.ts

```
@@ -245,4 +251,162 @@ export default async function transactions(app: FastifyInstance) {
       reply.send(result.rows[0]);
     },
   );
+
+  app.get<{
+    Reply: RealexPaymentObject | Error;
+    Querystring: RealexPaymentObjectQueryParams;
+  }>(
+    "/realex/paymentObject",
+    {
+      preValidation: app.verifyUser,
+      schema: {
+        tags: ["Transactions"],
+        querystring: RealexPaymentObjectQueryParams,
+        response: {
+          200: RealexPaymentObject,
+          404: HttpError,
+          422: HttpError,
+        },
+      },
+    },
+    async (request, reply) => {
+      const userId = request.user?.id;
+      const { providerId, amount, intentId } = request.query;
+
+      let providerRes;
+      try {
+        providerRes = await app.pg.query(
+          `
+          SELECT
+            provider_data as data,
+            status
+          FROM payment_providers
+          WHERE provider_id = $1
+          AND user_id = $2
+          `,
+          [providerId, userId],
+        );
+      } catch (err) {
+        app.log.error((err as Error).message);
+      }
+
+      if (!providerRes?.rows.length)
+        throw app.httpErrors.notFound("Provider not found");
+
+      const provider = providerRes.rows[0];
+      if (provider.status !== "connected") {
+        throw app.httpErrors.unprocessableEntity("Provider is not enabled");
+      }
+
+      const providerSecretsHandler = providerSecretsHandlersFactory("realex");
+      const { merchantId, sharedSecret } =
+        providerSecretsHandler.getClearTextData(
+          provider.data,
+        ) as unknown as RealexData;
+
+      const currency = "EUR";
+      const url =
+        process.env.REALEX_PAYMENT_URL ??
```

For now I think it's fine if we default to sandbox url - let me know if you prefer to throw an error if the env var is missing

---

I'm just concerned that we forget (maybe 6 months from now) to add that var and then suddenly we use a wrong url instead of throwing an error (which is much more visible). 

In case that's possible (maybe in a separate PR) we should have a validation with all the necessary env variables. What do you think? If a variable is required, we should throw an error on the startup

---

I agree. Let's address this separately